handler/src/test/java/io/netty/handler/ssl/CipherSuiteCanaryTest.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2018 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License, version
+ * 2.0 (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.netty.handler.ssl;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.ChannelPipeline;
+import io.netty.channel.DefaultEventLoopGroup;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.local.LocalAddress;
+import io.netty.channel.local.LocalChannel;
+import io.netty.channel.local.LocalServerChannel;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import io.netty.handler.ssl.util.SelfSignedCertificate;
+import io.netty.util.ReferenceCountUtil;
+import io.netty.util.concurrent.Promise;
+
+import java.net.SocketAddress;
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import org.junit.AfterClass;
+import org.junit.Assume;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+/**
+ * The purpose of this unit test is to act as a canary and catch changes in supported cipher suites.
+ */
+public class CipherSuiteCanaryTest {
+
+    private static EventLoopGroup GROUP;
+
+    private static SelfSignedCertificate CERT;
+
+    @BeforeClass
+    public static void init() throws Exception {
+        GROUP = new DefaultEventLoopGroup();
+        CERT = new SelfSignedCertificate();
+    }
+
+    @AfterClass
+    public static void destory() {
+        CERT.delete();
+        GROUP.shutdownGracefully();
+    }
+
+    @Test
+    public void testDiffieHellmanKeyExchange() throws Exception {
+        usingCipher("DHE-RSA-AES128-GCM-SHA256");
+    }
+
+    private static void usingCipher(final String cipher) throws Exception {
+        Assume.assumeTrue(OpenSsl.isAvailable());
+        assertTrue(OpenSsl.isCipherSuiteAvailable(cipher));
```

Comment:
I think this should be an `assumeTrue` as we also run our testsuite with `BoringSSL` for example